libs/features/personalization/personalization.js

```diff
@@ -271,30 +272,98 @@ function parsePlaceholders(placeholders, config, selectedVariantName = '') {
   return config;
 }
 
-function getPersonalizationVariant(manifestPath, variantNames = [], variantLabel = null) {
+const fetchEntitlements = async () => {
+  const [{ default: getUserEntitlements }] = await Promise.all([
+    import('../../blocks/global-navigation/utilities/getUserEntitlements.js'),
```

Comment:
There's one caveat with entitlements: the IMS client ID needs to be subscribed to JIL API. Maybe we should add some documentation to ensure all teams are aware of this step.